storage/src/main/java/tech/pegasys/artemis/storage/ChainStorageServer.java

```diff
@@ -47,14 +44,8 @@ public ChainStorageServer(EventBus eventBus, ArtemisConfiguration config) {
 
   @Subscribe
   public void onStoreDiskUpdate(final StoreDiskUpdateEvent event) {
-    try {
-      database.insert(event);
-
-      eventBus.post(new StoreDiskUpdateCompleteEvent(event.getTransactionId(), Optional.empty()));
-    } catch (final RuntimeException e) {
-      LOG.debug("Transaction " + event.getTransactionId() + " failed", e);
-      eventBus.post(new StoreDiskUpdateCompleteEvent(event.getTransactionId(), Optional.of(e)));
-    }
+    final DatabaseUpdateResult result = database.update(event);
+    eventBus.post(new StoreDiskUpdateCompleteEvent(event.getTransactionId(), result));
```

Comment:
It might be a good idea to switch EventChannels here

Reply:
👍We've got a ticket logged to do that conversion, so we can follow up with that refactor in a separate PR